com.unity.sg2/Editor/GraphUI/EditorCommon/CommandStateObserver/ShaderGraphCommandOverrides.cs

```diff
@@ -148,6 +143,59 @@ public static class ShaderGraphCommandOverrides
             }
         }
 
+        public static void HandleDeleteBlackboardItems(
+            UndoStateComponent undoState,
+            GraphModelStateComponent graphModelState,
+            SelectionStateComponent selectionState,
+            PreviewManager previewManager,
+            DeleteElementsCommand command)
+        {
+            var modelsToDelete = command.Models.ToList();
+
+            if (!modelsToDelete.Any())
+                return;
+
+            var graphModel = (ShaderGraphModel)graphModelState.GraphModel;
+
+            // Update previews
+            foreach (var model in modelsToDelete)
+            {
+                switch (model)
+                {
+                    case GraphDataVariableDeclarationModel variableDeclarationModel:
+
+                        // Gather all variable nodes linked to this blackboard item
+                        var linkedVariableNodes = graphModel.GetLinkedVariableNodes(variableDeclarationModel.graphDataName);
+                        foreach (var linkedVariableNode in linkedVariableNodes)
+                        {
+                            var graphDataVariableNode = linkedVariableNode as GraphDataVariableNodeModel;
+                            // Notify downstream nodes to update previews
+                            previewManager.OnNodeFlowChanged(graphDataVariableNode.graphDataName);
+                        }
+                        break;
+                }
+            }
+
+            // Delete GTF data and linked edges, variable nodes
+            DeleteElementsCommand.DefaultCommandHandler(undoState, graphModelState, selectionState, command);
+
+            using (var selectionStateUpdater = selectionState.UpdateScope)
+            {
+                selectionStateUpdater.SelectElements(modelsToDelete, false);
+            }
+
+            // Remove CLDS data
+            foreach (var model in modelsToDelete)
+            {
+                switch (model)
+                {
+                    case GraphDataVariableDeclarationModel variableDeclarationModel:
+                        graphModel.GraphHandler.RemoveReferableEntry(variableDeclarationModel.contextNodeName, variableDeclarationModel.graphDataName);
+                        break;
+                }
+            }
```

Comment:
Looks like the models are queued for deletion but not done so until the following frame, or so-- This is a bug for GTF. Models should stop updating or have view updates after they are deleted.